src/audio/crossover/crossover_generic.c

```diff
@@ -225,32 +225,38 @@ static void crossover_s32_default(struct comp_data *cd,
 				  int32_t num_sinks,
 				  uint32_t frames)
 {
+	/* Array to hold active sink streams; initialized to null */
+	struct audio_stream *sink_stream[SOF_CROSSOVER_MAX_STREAMS] = { NULL };
 	struct crossover_state *state;
 	const struct audio_stream *source_stream = bsource->data;
-	struct audio_stream *sink_stream;
 	int32_t *x, *y;
 	int ch, i, j;
 	int idx;
+	int active_sinks = 0;	 /* Counter for active sink streams */
 	int nch = audio_stream_get_channels(source_stream);
 	int32_t out[num_sinks];
 
+	/* Identify active sinks, avoid processing null sinks later */
+	for (j = 0; j < num_sinks; j++) {
```

Comment:
your commit message says "adding comments" but you are still changing code here

Reply:
corrected the commit message.